Sources/AlgoliaSearchClient/Client/Search/SearchClient+Wait.swift

```diff
@@ -0,0 +1,88 @@
+   
+    - parameter taskID: of the indexing [Task].
+    - parameter requestOptions: Configure request locally with [RequestOptions]
+  */
+  @discardableResult func taskStatus(for taskID: TaskID,
+                                     requestOptions: RequestOptions? = nil,
+                                     completion: @escaping  ResultCallback<TaskInfo>) -> Operation & TransportTask {
+    let command = Command.Advanced.TaskStatus(taskID: taskID, requestOptions: requestOptions)
+    return execute(command, completion: completion)
+  }
+
+  /**
+    Check the current TaskStatus of a given Task.
+   
+    - parameter taskID: of the indexing [Task].
+    - parameter requestOptions: Configure request locally with [RequestOptions]
+  */
+  @discardableResult func taskStatus(for taskID: TaskID,
+                                     requestOptions: RequestOptions? = nil) throws -> TaskInfo {
+    let command = Command.Advanced.TaskStatus(taskID: taskID, requestOptions: requestOptions)
+    return try execute(command)
+  }
+
+  // MARK: - Wait task
+
+  /**
+    Wait for a Task to complete before executing the next line of code, to synchronize index updates.
+    All write operations in Algolia are asynchronous by design.
+    It means that when you add or update an object to your index, our servers will reply to your request with
+    a TaskID as soon as they understood the write operation.
+    The actual insert and indexing will be done after replying to your code.
+    You can wait for a task to complete by using the TaskID and this method.
+   
+    - parameter taskID: of the indexing task to wait for.
+    - parameter requestOptions: Configure request locally with RequestOptions
+  */
+  @discardableResult func waitTask(withID taskID: TaskID,
+                                   timeout: TimeInterval? = nil,
+                                   requestOptions: RequestOptions? = nil,
+                                   completion: @escaping ResultCallback<TaskStatus>) -> Operation {
+    let task = WaitTask(client: self,
+                        taskID: taskID,
+                        timeout: timeout,
+                        requestOptions: requestOptions,
+                        completion: completion)
+    return launch(task)
+  }
+
+  /**
+    Wait for a Task to complete before executing the next line of code, to synchronize index updates.
+    All write operations in Algolia are asynchronous by design.
+    It means that when you add or update an object to your index, our servers will reply to your request with
+    a TaskID as soon as they understood the write operation.
+    The actual insert and indexing will be done after replying to your code.
+    You can wait for a task to complete by using the TaskID and this method.
+   
+    - parameter taskID: of the indexing task to wait for.
+    - parameter requestOptions: Configure request locally with RequestOptions
+  */
+  @discardableResult func waitTask(withID taskID: TaskID,
```

Comment:
Maybe a `DictionaryTaskID` to avoid passing index `TaskID`s ?

Reply:
I'm not sure if it makes sense to separate `DictionaryTaskID`. I suppose, another waitable tasks may be introduced later on the application level. But it's a good idea to separate `IndexTaskID` from `ApplicationTaskID` to avoid their misuse. 
What do you think?